### PR TITLE
fixed image deletion bug, when profile settings are saved.

### LINKED ISSUE #504: 🐛 BUG: profile picture will be deleted
When I save my profile, my profile picture will be deleted.

### PR DESCRIPTION
**Pull request by Gargamil**
## Description:

profile image will now not be deleted, if user changes his profile settings

Closes #504 

## Improvements
- 

## Known Issues:
_NONE_

------
_Remember to prefix your PR-Title with `⚠ WIP: ` if you still work on it.
If you have reached a final state, remove the prefix._

### CODE REVIEW

**Comment by PatrickSkowronek** [comment]
Your build failed. One test failed, because you need to change the expected value in the test. There are many issues in the Frontend. Did you run it?

ERROR in src/app/course/course-edit/course-user-list/course-user-list.component.ts(80,21): error TS2345: Argument of type 'User' is not assignable to parameter of type 'IUser'.
src/app/course/course-edit/course-user-list/course-user-list.component.ts(82,45): error TS2345: Argument of type 'User' is not assignable to parameter of type 'IUser'.
src/app/course/course-edit/course-user-list/course-user-list.component.ts(88,43): error TS2345: Argument of type 'User' is not assignable to parameter of type 'IUser'.
src/app/course/course-edit/general-tab/general-tab.component.ts(163,83): error TS2345: Argument of type 'User' is not assignable to parameter of type 'IUser'.
src/app/models/User.ts(5,14): error TS2420: Class 'User' incorrectly implements interface 'IUser'.
  Types of property 'profile' are incompatible.
    Type '{ firstName: string; lastName: string; picture: IFile; theme: string; }' is not assignable to type '{ firstName: string; lastName: string; picture: IFile; theme: string; }'. Two different types with this name exist, but they are unrelated.
      Types of property 'picture' are incompatible.
        Type 'IFile' is not assignable to type 'IFile'. Two different types with this name exist, but they are unrelated.
          Property 'path' is missing in type 'IFile'.
src/app/models/User.ts(19,5): error TS90010: Type '{ firstName: string; lastName: string; picture: IFile; theme: string; }' is not assignable to type '{ firstName: string; lastName: string; picture: IFile; theme: string; }'. Two different types with this name exist, but they are unrelated.
  Types of property 'picture' are incompatible.
    Type 'IFile' is not assignable to type 'IFile'. Two different types with this name exist, but they are unrelated.
      Property '_id' is optional in type 'IFile' but required in type 'IFile'.
src/app/notification/notification.component.ts(34,54): error TS2345: Argument of type 'User' is not assignable to parameter of type 'IUser'.
src/app/shared/utils/LastVisitedCourseContainerUpdater.ts(15,29): error TS2345: Argument of type 'User' is not assignable to parameter of type 'IUser'.
src/app/user/user-details/user-details.component.ts(52,9): error TS2322: Type 'User' is not assignable to type 'IUser'.
src/app/user/user-settings/user-settings.component.ts(54,92): error TS2345: Argument of type 'User' is not assignable to parameter of type 'IUser'.
  Types of property 'profile' are incompatible.
    Type '{ firstName: string; lastName: string; picture: IFile; theme: string; }' is not assignable to type '{ firstName: string; lastName: string; picture: IFile; theme: string; }'. Two different types with this name exist, but they are unrelated.
      Types of property 'picture' are incompatible.
        Type 'IFile' is not assignable to type 'IFile'. Two different types with this name exist, but they are unrelated.
src/app/user/user-settings/user-settings.component.ts(90,97): error TS2345: Argument of type 'User' is not assignable to parameter of type 'IUser'.``

**Comment by PatrickSkowronek** [comment]
And please the the PR to "WIP: " so we can see that you work on it!

**Comment by Gargamil** [comment]
The Problem is: To get this working, IUser must the "old" IFile, or the Database has to be migrated to the new IFile representation. 
With my current knowledge i really don't want to do such migration.. 

So whats to do here? 
Of course i can use a non typed representation of the File data.. But i really don't want that..

**Comment by grosspersky** [comment]
Try fixing the frontend model of the user interface too:
https://github.com/h-da/geli/blob/develop/app/webFrontend/src/app/models/User.ts

**Comment by PatrickSkowronek** [comment]
Your changes work. But one test fails, because you changed the IFile model. Please change the testcase than this is done.

File: app/webFrontend/src/app/user/user-edit/user-edit.component.ts [hunk 1]
```diff
@@ -77,29 +77,25 @@ export class UserEditComponent implements OnInit {
...
-        saveIUser[key] = userFormModel[key];
-      }
-    }
-    for (const key in this.user) {
-      if (typeof saveIUser[key] === 'undefined') {
-        saveIUser[key] = this.user[key];
-      }
+    const userFormModel = this.userForm.value
+
+    if(this.user.profile.picture) {
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/user/user-edit/user-edit.component.ts]: Here is your linting error!

Comment (Gargamil) [reply app/webFrontend/src/app/user/user-edit/user-edit.component.ts]: yea, already seen this.. Annoying. Fixed now.
Sorry for inconvenience. In future i will do all these checks first locally.

**Pull request merged**